# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../qtparted/parted-1.7-fix.patch
# Copyright (C) 2006 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./src/qp_fswrap.cpp.orig	2005-07-07 00:29:54.000000000 +0300
+++ ./src/qp_fswrap.cpp	2006-12-15 05:25:20.281348500 +0200
#Patch from Gentoo kevquinn
@@ -1002,7 +1002,11 @@
 /*---FAT32 WRAPPER---------------------------------------------------------------*/
 QString QP_FSFat32::_get_label(PedPartition *part) 
 {
-    char buffer[PED_SECTOR_SIZE];
+#ifdef PED_SECTOR_SIZE // PED_SECTOR_SIZE is gone in parted 1.7.x
+    char *buffer=new char[PED_SECTOR_SIZE];
+#else
+    char *buffer=new char[part->disk->dev->sector_size];
+#endif
     char label[12];
     
     if (!QP_FSWrap::read_sector(part, 0, 1, buffer)) 
